numpy/lib/tests/test_stride_tricks.py

```diff
@@ -333,6 +336,69 @@ def test_as_strided():
     assert_equal(a.dtype, a_view.dtype)
     assert_array_equal([r] * 3, a_view)
 
+
+def test_sliding_window_view():
```

Comment:
I think it would be nice to split this into smaller tests. One thing you can do is wrap the tests in a class,

```
class TestSlidingWindowView:
    def test_1d(self): ...

    def test_2d(self):

    def test_errors(self):
```
Shorter tests are easier to follow, as you don't have to keep track of local variables for as long.

Up to you how you want to split these - every time you start with a new array would be a reasonable choice.